components/script/webdriver_handlers.rs

```diff
@@ -81,8 +89,46 @@ pub unsafe fn jsval_to_webdriver(cx: *mut JSContext, val: HandleValue) -> WebDri
                 _ => unreachable!(),
             };
         Ok(WebDriverJSValue::String(String::from(string)))
-    } else if val.get().is_null() {
-        Ok(WebDriverJSValue::Null)
+    } else if val.get().is_object() {
+        rooted!(in(cx) let object = match FromJSValConvertible::from_jsval(cx, val, ()).unwrap() {
+            ConversionResult::Success(object) => object,
+            _ => unreachable!(),
+        });
+        let _ac = JSAutoRealm::new(cx, *object);
+
+        if !is_array_like(cx, val) {
+            return Err(WebDriverJSError::UnknownType);
+        }
+
+        let mut result: Vec<WebDriverJSValue> = Vec::new();
+
+        let length =
+            match get_property::<u32>(cx, object.handle(), "length", ConversionBehavior::Default) {
+                Ok(length) => match length {
+                    Some(length) => length,
+                    Err(error) => {
```

Comment:
I do not believe there is an enum with `Some/Err` variants ;)

Reply:
Ups, looks like I was too sleepy when I wrote that :D